components/brave_ads/core/internal/common/net/http/http_status_code_util.cc

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/common/net/http/http_status_code_util.h"
+
+#include "base/containers/fixed_flat_set.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr auto kAllowedHttpStatusCodes = base::MakeFixedFlatSet<int>({
+    400,  // Bad Request.
+    401,  // Unauthorized.
+    403,  // Forbidden.
+    404,  // Not Found.
+    408,  // Request Timeout.
+    429,  // Too Many Requests.
+    451,  // Unavailable For Legal Reasons.
+    500,  // Internal Server Error.
+    502,  // Bad Gateway.
+    503,  // Service Unavailable.
+    504   // Gateway Timeout.
+});
+
+std::string HttpStatusCodeClassToString(const int http_status_code_class) {
+  return base::StringPrintf("%dxx", http_status_code_class);
```

Comment:
Will not happen in practice, but unless we `CHECK()` here, it is possible for clients to pass nonsensical values and so we should do _something_.
```suggestion
  if (http_status_code_class < 100 || http_status_code_class > 599) {
    return "0xx"
  }
  return base::StringPrintf("%dxx", http_status_code_class);
```

Reply:
Warning: may not compile, I typed this in the GitHub PR UI ;)

Reply:
Added but slightly different solution, thanks